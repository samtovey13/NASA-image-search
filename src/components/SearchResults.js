import '../styles/SearchResults.css';
import React from 'react';
import PropTypes from 'prop-types';

const SearchResults = ({results}) => {
  if ((results.length === 1 && results[0] === 404) || (!results.length)) {
    return <p className="error-message">No results</p>
  } else if (results.length === 1 && results[0] === 500) {
    return <p className="error-message">Server error, try again later</p>
  } else if (results.length === 1 && results[0] === 400) {
    return <p className="error-message">Invalid search term</p>
  } else {
    return <>
      <p className="search-results-header">Search Results</p>
      <div className="search-results-images" data-testid="search-results-images">
        {
          results.map((image, index) => (
            <span className="image-wrapper" key={`image-wrapper-${index}`}><img
              className="results-image"
              src={image} 
              alt="NASA" 
              key={`result-${index}`}
              data-testid={`image-result-${index}`}
            /></span>
          ))
        }
        <span className="last-child"></span>
      </div>
    </>
  }
}

SearchResults.propTypes = {
  results: PropTypes.array
}

export default SearchResults;