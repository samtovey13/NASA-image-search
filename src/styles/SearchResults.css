.search-results-images {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: space-between;
  margin: 10px;
}

.image-wrapper {
  height: 300px;
  flex-grow: 0;
  margin: 10px 2px;
}

.results-image {
  max-width: 100%;
  max-height: 100%;
  object-fit: cover;
  display: block;
  margin: auto;

}
.last-child {
  flex-grow: 4;
}


@media screen and (max-width: 900px) {
  .image-wrapper {
    height: 200px;
  }
}

@media screen and (max-width: 600px) {
  .search-results-images {
    display:block;
  }
  .results-image {
    width: 100%;
    display: block;
    margin: auto;
  }
}