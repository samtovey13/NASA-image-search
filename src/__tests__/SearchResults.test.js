import SearchResults from '../components/SearchResults';
import React from 'react';
import { render } from '@testing-library/react';

const mockSearchResults = [
  "image1.jpg",
  "image2.jpg",
  "image3.jpg"
];

describe("SearchResults", () => {
  it ("renders correctly with no results", () => {
    const { asFragment } = render( <SearchResults results={[]}/>);

    expect(asFragment()).toMatchSnapshot();
  });

  it ("renders correctly with results", () => {
    const { asFragment } = render( <SearchResults results={mockSearchResults}/>);

    expect(asFragment()).toMatchSnapshot();
  });

  it ("renders the correct components", () => {
    const { getByTestId, getByText } = render(<SearchResults results={mockSearchResults}/>);

    expect(getByText("Search Results")).toHaveClass("search-results-header");
    expect(getByTestId("search-results-images")).toBeInTheDocument();
    expect(getByTestId("image-result-0")).toBeInTheDocument();
    expect(getByTestId("image-result-1")).toBeInTheDocument();
    expect(getByTestId("image-result-2")).toBeInTheDocument();
  });

  describe ("handles bad requests to the API", () => {
    it("handles a 400 response", () => {
      const {getByText} = render(<SearchResults results={[400]} />);
      expect(getByText("Invalid search term")).toHaveClass("error-message");
    });

    it("handles a 404 response", () => {
      const {getByText} = render(<SearchResults results={[404]} />);
      expect(getByText("No results")).toHaveClass("error-message");
    });
    
    it("handles a 500 response", () => {
    const {getByText} = render(<SearchResults results={[500]} />);
    expect(getByText("Server error, try again later")).toHaveClass("error-message");      
    })
  })
});