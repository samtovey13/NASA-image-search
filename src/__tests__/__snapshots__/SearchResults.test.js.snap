// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p
    class="search-results-header"
  >
    Search Results
  </p>
  <div
    class="search-results-images"
    data-testid="search-results-images"
  >
    <span
      class="image-wrapper"
    >
      <img
        alt="NASA"
        class="results-image"
        data-testid="image-result-0"
        src="image1.jpg"
      />
    </span>
    <span
      class="image-wrapper"
    >
      <img
        alt="NASA"
        class="results-image"
        data-testid="image-result-1"
        src="image2.jpg"
      />
    </span>
    <span
      class="image-wrapper"
    >
      <img
        alt="NASA"
        class="results-image"
        data-testid="image-result-2"
        src="image3.jpg"
      />
    </span>
    <span
      class="last-child"
    />
  </div>
</DocumentFragment>
`;
